vpr/src/place/place_constraints.cpp

```diff
@@ -257,3 +258,95 @@ void load_cluster_constraints() {
         }
     }
 }
+
+void mark_fixed_blocks() {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.mutable_placement();
+    auto& floorplanning_ctx = g_vpr_ctx.floorplanning();
+
+    for (auto blk_id : cluster_ctx.clb_nlist.blocks()) {
+        if (!is_cluster_constrained(blk_id)) {
+            continue;
+        }
+        PartitionRegion pr = floorplanning_ctx.cluster_constraints[blk_id];
+        auto block_type = cluster_ctx.clb_nlist.block_type(blk_id);
+        t_pl_loc loc;
+
+        if (is_pr_size_one(pr, block_type, loc)) {
+            //Set block location and grid usage
+            set_block_location(blk_id, loc);
+
+            //Set as fixed
+            place_ctx.block_locs[blk_id].is_fixed = true;
+        }
+    }
+}
+
+bool is_region_size_one(const Region& reg, t_logical_block_type_ptr block_type) {
```

Comment:
could also loop through all grid locations in the region to see whether it covers 0, 1, or more than one tile. This would also then cover the case where a bigger region only essentially covers one legal loc

Reply:
given this case -> could also just return int and have 0, 1, 2 as possible return values -> make sure this is commented very clearly in this case